utils/src/treev2/quickPickWizard/quickPickAzureResource/QuickPickAppResourceStep.ts

```diff
@@ -26,29 +27,32 @@ export class QuickPickAppResourceStep extends GenericQuickPickStep<AzureResource
         return pickedAppResource;
     }
 
-    protected isDirectPick(node: TreeItem): boolean {
-        // If childItemFilter is defined, this cannot be a direct pick
-        if (this.pickOptions.childItemFilter) {
-            return false;
-        }
+    readonly pickFilter: PickFilter = new AppResourcePickFilter(this.pickOptions);

```

Comment:
Since this isn't being called within the constructor, I'm not 100% sure that `this.pickOptions` will be available. I don't know the order for sure but I thought that class-scoped initializers like this get called _before_ the constructor.

Reply:
The constructor is compiled to the following JS. Seems like property initializers are appended to the constructor body, so it works.

```js
class QuickPickGroupStep extends GenericQuickPickStep_1.GenericQuickPickStep {
    constructor(tdp, options) {
        super(tdp, Object.assign(Object.assign({}, options), { skipIfOne: true }));
        this.pickFilter = new GroupPickFilter(this.pickOptions);
    }
}
```